gc/gc-base/src/main/java/org/projectnessie/gc/base/DistributedIdentifyContents.java

```diff
@@ -46,50 +47,83 @@ public DistributedIdentifyContents(SparkSession session, GCParams gcParams) {
   }
 
   /**
-   * Compute the bloom filter per content id by walking all the live references in a distributed way
-   * using spark.
+   * Compute the bloom filter by walking all the live references in a distributed way using spark.
    *
    * @param references list of all the references (JSON serialized)
    * @param bloomFilterSize size of bloom filter to be used
    * @param droppedRefTimeMap map of dropped time for reference@hash (JSON serialized)
-   * @return map of {@link ContentBloomFilter} per content-id.
+   * @param runId current run id of the gc
+   * @param startedAt gc start time
+   * @return {@link BloomFilter} object.
    */
-  public Map<String, ContentBloomFilter> getLiveContentsBloomFilters(
-      List<String> references, long bloomFilterSize, Map<String, Instant> droppedRefTimeMap) {
-    IdentifyContentsPerExecutor executor = new IdentifyContentsPerExecutor(gcParams);
-    List<Map<String, ContentBloomFilter>> bloomFilterMaps =
+  public BloomFilter<Content> getLiveContentsBloomFilter(
+      List<String> references,
+      long bloomFilterSize,
+      Map<String, Instant> droppedRefTimeMap,
+      String runId,
+      Timestamp startedAt) {
+    IdentifyLiveContentsPerExecutor executor = new IdentifyLiveContentsPerExecutor(gcParams);
+    List<LiveContentsResult> liveContentsResults =
```

Comment:
Can't pass directly `JavaRDD` because I need to collect bloom filter and merge it along with preparing checkpoint rows. If it was just checkpoint rows, it would have been possible to pass `JavaRDD` to the `createDataFrame()`

So, I will refactor only other parts (removing double traversal and extra lists)